core/src/main/java/org/elasticsearch/search/rescore/RescoreParseElement.java

```diff
@@ -71,7 +74,7 @@ public void parseSingleRescoreContext(XContentParser parser, SearchContext conte
         if (windowSize != null) {
             rescoreContext.setWindowSize(windowSize.intValue());
         }
-        context.addRescore(rescoreContext);
+        return rescoreContext;
```

Comment:
Why does this class need to change? the workings of this class should not need to change since it is only used on the shards and will be removed when the search refactoring is completed. I think we should leave it as it was so we have a reference to how the functionality used to be in case we need to refer to it.